static/src/javascripts/projects/commercial/modules/dfp/empty-advert.js

```diff
@@ -1,14 +1,24 @@
 // @flow
 
-/* eslint no-multi-assign: "off"*/
-
 import fastdom from 'fastdom';
+import { Advert } from 'commercial/modules/dfp/Advert';
+import dfpEnv from 'commercial/modules/dfp/dfp-env';
+
+const removeFromDfpEnv = advert => {
+    const removeAdvert = (adverts: Array<Advert>): Array<Advert> =>
+        adverts.filter(_ => _ !== advert);
+
+    dfpEnv.adverts = removeAdvert(dfpEnv.adverts);
+    dfpEnv.advertsToRefresh = removeAdvert(dfpEnv.advertsToRefresh);
+    dfpEnv.advertsToLoad = removeAdvert(dfpEnv.advertsToLoad);
+    delete dfpEnv.advertIds[advert.id];
```

Comment:
Always avoid `delete`, it has a dramatic impact on the heap. But even though, this is false. Because you are rebuilding `adverts` from scratch, all the indices in `advertIds` will be pointing to something else.